File: src/main/java/zw/co/hisolutions/core/service/AudianceService.java
package zw.co.hisolutions.core.service;
  
import zw.co.hisolutions.common.service.GenericService;
import zw.co.hisolutions.core.entity.Audiance;

public interface AudianceService extends GenericService<Audiance, Long>{
    public Audiance findByName(String name);   
}

File: frontend/src/app/shared/services/rest/audiance.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {DataService} from 'shared/services/data.service';

@Injectable()
export class AudianceService extends DataService  {

    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/audiances'; 
        super(httpClient, resourcePath);
    }

}
